mobile/examples/phi-3/android/app/src/main/java/ai/onnxruntime/genai/demo/MainActivity.java

```diff
@@ -84,9 +108,44 @@ public void onClick(View v) {
                 new Thread(new Runnable() {
                     @Override
                     public void run() {
+                        TokenizerStream stream = null;
+                        GeneratorParams generatorParams = null;
+                        Generator generator = null;
+                        Sequences encodedPrompt = null;
                         try {
-                            genAIWrapper.run(promptQuestion_formatted);
-                        } catch (GenAIException e) {
+                            stream = tokenizer.createStream();
+
+                            generatorParams = model.createGeneratorParams();
+                            //examples for optional parameters to format AI response
+                            //generatorParams.setSearchOption("length_penalty", 1000);
+                            //generatorParams.setSearchOption("max_length", 500);
+
+                            encodedPrompt = tokenizer.encode(promptQuestion_formatted);
+                            generatorParams.setInput(encodedPrompt);
+
+                            generator = new Generator(model, generatorParams);
+
+                            while (!generator.isDone()) {
+                                generator.computeLogits();
+                                generator.generateNextToken();
+                 
+                                int token = generator.getLastTokenInSequence(0);
+                 
+                                tokenListener.accept(stream.decode(token));
+                            }
```

Comment:
Do we need to show all steps here, or would the simpler iteration pattern be better

```
            Generator generator = new Generator(model, generatorParams)) {
          // iterate (which calls computeLogits, generateNextToken, getLastTokenInSequence and
          // isDone)
          for (int token_id : generator) {
            // decode and call listener
            String token = stream.decode(token_id);
            listener.accept(token);
          }
```

Reply:
I did this for readability since this is meant to be a tutorial for mobile developers.